### ISSUE TITLE
kendo-react-data-tools throws error on build when node_modules type checking is enabled

### ISSUE THREAD

**Issue by Mako217** [telerik/kendo-react#2363]
-
  Please help us process GitHub issues faster by providing the necessary information in the template format that follows.
  We might not be able to assist with issues which lack important information.
-->

## I'm submitting a...
* Bug report

## Current behavior
When trying to build a project, an error: "Type parameter defaults can only reference previously declared type parameters" is thrown in `index.d.ts` file of `kendo-react-data-tools `. This is caused by generic types being named the same as the default types they are supposed to take, for example:
```
export declare interface TreeColumnBaseProps<CellProps = CellProps, HeaderCellProps = HeaderCellProps, FilterCellProps = FilterCellProps> extends ColumnBaseProps<CellProps> {
```
This fails the build of the project, because we have type checking enabled in node_modules

## Expected behavior
Code builds properly with node_modules type checking enabled. To achieve this the generic types need to be named differently than default interfaces. For example:
```
export declare interface TreeColumnBaseProps<C = CellProps, H = HeaderCellProps, F = FilterCellProps> extends ColumnBaseProps<C> {
```

## Minimal reproduction of the problem with instructions
1. Have `kendo-react-data-tools` installed with version 7.3.0 or higher.
2. Have type checking enabled in `tsconfig.json` (`"skipLibCheck": false`)
3. Try to build the project

## What is the motivation or use case for changing the behavior?
I want to have type checking enabled in the project, which now is not possible because `kendo-react-data-tools` code fails.

## Environment

Package versions:

```
@progress/kendo-react-data-tools@8.1.1
```

Browser:

Irrelevant

System:
- TypeScript version: 5.2.2
- Node version: v20.10.0
- Platform:  Windows 10

I have checked which version this appeared in and it's 7.3.0, also it might be good to include it in the changelog so that people don't have to investigate if there is a problem on their end.

**Comment by silviyaboteva** [comment]
Hello @Mako217,

We reproduced the issue on our end as well. We consider this as a bug and will be fixed as soon as possible. 
Thank you for bringing this to our attention.

**Comment by silviyaboteva** [comment]
The fix is included in version `8.2.0-develop.13` and will be available in the next official version of the package.
